### PR TITLE
Rework on resource deallocation in switchapi TDI API

### PR DESCRIPTION
**Pull request by 5abeel**
Signed-off-by: Sabeel Ansari <sabeel.ansari@intel.com>

### CODE REVIEW

**Comment by ipdksw** [comment]
Result:SUCCESS JenkinsConsole:https://s3.ipdk.io/ovs/PR-42-1/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/6e7761f652a4d20107ef76326e0c10d210cde167

File: p4proto/kctrl/switchapi/switch_pd_utils.h [hunk 1]
```diff
@@ -57,10 +57,22 @@ bf_status_t switch_pd_deallocate_handle_session(bf_rt_table_key_hdl *key_hdl_t,
```
Comment (nupuruttarwar) [inline p4proto/kctrl/switchapi/switch_pd_utils.h]: If these are not being used, we can remove the declarations

**Review by nupuruttarwar — CHANGES_REQUESTED** [review]

**Comment by ipdksw** [comment]
Result:SUCCESS JenkinsConsole:https://s3.ipdk.io/ovs/PR-42-2/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/6e7761f652a4d20107ef76326e0c10d210cde167

File: p4proto/kctrl/switchapi/switch_pd_utils.h (hunk 1 above)
Comment (5abeel) [reply p4proto/kctrl/switchapi/switch_pd_utils.h]: Removed

**Comment by ipdksw** [comment]
Result:SUCCESS JenkinsConsole:https://s3.ipdk.io/ovs/PR-42-3/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/eba73dad932bc7f89b1b0098ee360877bdc02925

File: p4proto/kctrl/switchapi/switch_pd_utils.h [hunk 2]
```diff
@@ -50,17 +50,24 @@ bf_status_t switch_pd_allocate_handle_session(const bf_dev_id_t device_id,
                                               bf_rt_info_hdl **bfrt_info_hdl_t,
                                               bf_rt_session_hdl **session_t);
```
Comment (nupuruttarwar) [inline p4proto/kctrl/switchapi/switch_pd_utils.h]: We can remove this one as well right? Also, can you rebase the branch and take care of DCO check?

**Review by nupuruttarwar — CHANGES_REQUESTED** [review]

**Comment by ipdksw** [comment]
Result:SUCCESS JenkinsConsole:https://s3.ipdk.io/ovs/PR-42-4/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/eba73dad932bc7f89b1b0098ee360877bdc02925

File: p4proto/kctrl/switchapi/switch_pd_utils.h (hunk 2 above)
Comment (5abeel) [reply p4proto/kctrl/switchapi/switch_pd_utils.h]: Ack

**Comment by ipdksw** [comment]
Result:SUCCESS JenkinsConsole:https://s3.ipdk.io/ovs/PR-42-5/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/711361c3d5433c04447d09fa286757470e153f57

**Comment by ipdksw** [comment]
Result:SUCCESS JenkinsConsole:https://s3.ipdk.io/ovs/PR-42-6/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/711361c3d5433c04447d09fa286757470e153f57

**Comment by ipdksw** [comment]
Result:SUCCESS JenkinsConsole:https://s3.ipdk.io/ovs/PR-42-7/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/b2220e4c3482919bb1b9c953293444ca84f90689

**Comment by ipdksw** [comment]
Result:SUCCESS JenkinsConsole:https://s3.ipdk.io/ovs/PR-42-8/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/b2220e4c3482919bb1b9c953293444ca84f90689

**Comment by ipdksw** [comment]
Result:SUCCESS JenkinsConsole:https://s3.ipdk.io/ovs/PR-42-9/index.html GitCommit:https://github.com/ipdk-io/ovs/commits/b2220e4c3482919bb1b9c953293444ca84f90689

**Review by nupuruttarwar — APPROVED** [review]
Looks good to me

**Review by np2-git — APPROVED** [review]

**Pull request merged**